File: qc-core/src/main/java/com/eager/questioncloud/domain/user/UserInformation.java
package com.eager.questioncloud.domain.user;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class UserInformation {
    private String email;
    private String phone;
    private String name;
    private String profileImage;

    public static UserInformation guest = new UserInformation("guest", "guest", "guest", null);

    public static UserInformation create(CreateUser createUser) {
        return new UserInformation(createUser.getEmail(), createUser.getPhone(), createUser.getName(), null);
    }

    public static UserInformation getGuestInformation() {
        return guest;
    }

    public UserInformation updateUserInformation(String name, String profileImage) {
        return new UserInformation(this.email, this.phone, name, profileImage);
    }
}

File: qc-domain/qc-user/src/main/kotlin/com/eager/questioncloud/user/domain/UserInformation.kt
package com.eager.questioncloud.user.domain

import com.eager.questioncloud.user.dto.CreateUser

class UserInformation(
    var email: String,
    var phone: String,
    var name: String,
    var profileImage: String = "default",
) {
    fun updateUserInformation(name: String, profileImage: String): UserInformation {
        return UserInformation(this.email, this.phone, name, profileImage)
    }

    companion object {
        var guestInformation: UserInformation =
            UserInformation("guest", "guest", "guest", "guest")

        fun create(createUser: CreateUser): UserInformation {
            return UserInformation(
                email = createUser.email,
                phone = createUser.phone,
                name = createUser.name
            )
        }
    }
}
